tests/test_copy_move.cpp

```diff
@@ -214,6 +214,7 @@ TEST_SUBMODULE(copy_move_policies, m) {
     };
     py::class_<MoveIssue2>(m, "MoveIssue2").def(py::init<int>()).def_readwrite("value", &MoveIssue2::v);
 
-    m.def("get_moveissue1", [](int i) { return new MoveIssue1(i); }, py::return_value_policy::move);
+    // #2742: Don't expect ownership of raw pointer to `new`ed object to be transferred with `py::return_value_policy::move`
+    m.def("get_moveissue1", [](int i) { return std::unique_ptr<MoveIssue1>(new MoveIssue1(i)); }, py::return_value_policy::move);
```

Comment:
Just to make things clear, this is avoiding the issue reported in #2742, but not proposing a definite solution.

Moreover, the issue that was a precursor to adding this tests doesn't seem to depend on `get_moveissue1()` returning a raw owning pointer.

Reply:
No, but if we decide to have a different approach to #2742, we'll add new tests for those, anyway? So this change could be semi-permanent?